src/messages/MOSDOp.h

```diff
@@ -364,9 +364,38 @@ struct ceph_osd_request_head {
 
       encode(retry_attempt, payload);
       encode(features, payload);
-    } else {
-      // latest v8 encoding with hobject_t hash separate from pgid, no
+    } else if (!HAVE_FEATURE(features, SERVER_SQUID)) {
```

Comment:
Hmm, I guess this should be fine assuming no plans for backporting.

Reply:
no such plans